pkg/apis/camel/v1/integrationkit_types.go

```diff
@@ -110,6 +110,8 @@ const (
 	IntegrationKitPhaseReady IntegrationKitPhase = "Ready"
 	// IntegrationKitPhaseError --
 	IntegrationKitPhaseError IntegrationKitPhase = "Error"
+	// IntegrationKitPhaseCannotBuild
+	IntegrationKitPhaseCannotBuild IntegrationKitPhase = "Cannot Build"
```

Comment:
Could this be a reason of `IntegrationKitPhaseError`, instead of introducing a new phase?

Reply:
The reason for this is as per the commit message:

> "Cannot Build" differentiates from "Error" since error action can
> put the integration kit back to initialization phase to try and build
> again. This makes no sense for "Cannot Build" cannot build

Reply:
I see, though this only happens when the IntegrationKit digest changes, which I think makes it acceptable. Ideally, IntegrationKit would mirror the Build, which has a Failed phase for recoverable issues, and an Error phase for unrecoverable ones. Still I'd rather re-use the `IntegrationKitPhaseError` in that case, instead of introducing a very specific `IntegrationKitPhaseCannotBuild` phase. If you think that a phase for unrecoverable error is mandatory, then I would suggest mirroring the Build API.

Reply:
Sure I can change it to `IntegrationKitPhaseError`. It would simplify some of the tests I had to add in to consider `...CannotBuild`. I'll try it out and see how it pans out.